src/V3Ast.h

```diff
@@ -1247,6 +1247,45 @@ class VBasicTypeKey final {
     ~VBasicTypeKey() = default;
 };
 
+// ######################################################################
+//  VSelfPointerText - Represents text to be emitted before a given var reference, call, etc. to
+//  serve as a pointer to a 'self' object. For example, it could be empty (no self pointer), or the
+//  string 'this', or 'vlSymsp->...'
+
+class VSelfPointerText final {
+private:
+    // STATIC MEMBERS
+    // Keep these in shared pointers to avoid branching for special cases
+    static const std::shared_ptr<const string> s_emptyp;
+    static const std::shared_ptr<const string> s_thisp;
+
+    // MEMBERS
+    std::shared_ptr<const string> m_strp;
+
+    // CONSTRUCTORS
+    VSelfPointerText(std::shared_ptr<const string> strp)
+        : m_strp{strp} {}
+    VSelfPointerText(string&& str)
+        : m_strp{std::make_shared<const string>(std::move(str))} {}
+
+public:
+    // METHODS
+    bool isEmpty() const { return m_strp == s_emptyp; }
+    bool isVlSym() const { return m_strp->find("vlSymsp") != string::npos; }
+    bool hasThis() const { return m_strp == s_thisp || VString::startsWith(*m_strp, "this"); }
```

Comment:
If it does, then it would've broken without this change (the logic is the same). It's something that we should check, but outside the scope of this PR. I think in the future we should move towards a more structured representation of a self pointer (among other things) instead of using string comparisons, but that's a whole can of worms.